### PR TITLE
Dns functional tests

### LINKED ISSUE #13: dns func tests
implement functional tests for dns

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
Closes #13

### CODE REVIEW

**Review by outcatcher — DISMISSED** [review]

**Review by gtema — CHANGES_REQUESTED** [review]
really great work. There are however some issues:
- due to the use of non unique names (in zones) there is potential of conflicts when few PRs run tests at the same time.
- creating/deleting network for every test can become expensive (both time and potential of failures due to quota or throttling). You can try to create all required prerequisites once at start (do not use setUpClass please). https://github.com/opentelekomcloud/python-otcextensions/blob/master/otcextensions/tests/functional/osclient/rds/v3/test_instance.py used this approach for OSC testing (not the most optimal implementation, but as a concept)

File: otcextensions/tests/functional/sdk/dns/__init__.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            pass
+        self.client = self.conn.dns
+
+    def create_network(self):
+        self.cidr = '192.168.0.0/16'
+        self.ipv4 = 4
+        self.dns_nameservers = ['100.125.4.25', '8.8.8.8']
+
+        network = self.conn.network.create_network(name=self.net_name)
+        self.assertEqual(self.net_name, network.name)
```
Comment (Polina-Gubina) [inline otcextensions/tests/functional/sdk/dns/__init__.py]: it is not a thing which need to be tested here, is it neccessary?

**Review by Polina-Gubina — DISMISSED** [review]

File: otcextensions/tests/functional/sdk/dns/__init__.py (hunk 1 above)
Comment (anton-sidelnikov) [reply otcextensions/tests/functional/sdk/dns/__init__.py]: this is for ensure that resource successfully created

**Comment by anton-sidelnikov** [comment]
> really great work. There are however some issues:
> 
> * due to the use of non unique names (in zones) there is potential of conflicts when few PRs run tests at the same time.
> * creating/deleting network for every test can become expensive (both time and potential of failures due to quota or throttling). You can try to create all required prerequisites once at start (do not use setUpClass please). https://github.com/opentelekomcloud/python-otcextensions/blob/master/otcextensions/tests/functional/osclient/rds/v3/test_instance.py used this approach for OSC testing (not the most optimal implementation, but as a concept)

did some changes, added unique names, re-wrote zone tests

**Review by Polina-Gubina — DISMISSED** [review]

**Review by lego963 — DISMISSED** [review]

**Review by outcatcher — DISMISSED** [review]

**Review by irina-pereiaslavskaia — DISMISSED** [review]

**Review by lego963 — APPROVED** [review]

**Review by Polina-Gubina — APPROVED** [review]

File: otcextensions/tests/functional/sdk/dns/v2/test_zone.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            self.networks.append(network)
+
+    def test_01_list_zones(self):
+        self._create_zone(self.public_zone_alias)
+        self.all_zones = list(self.client.zones())
+        self.assertGreaterEqual(len(self.all_zones), 0)
+        if len(self.all_zones) > 0:
+            zone = self.all_zones[0]
+            zone = self.client.get_zone(zone=zone.id)
+            self.assertIsNotNone(zone)
```
Comment (Polina-Gubina) [inline otcextensions/tests/functional/sdk/dns/v2/test_zone.py:70]: is it right? can be zone none if it was returned in list of existing zones?

**Review by Polina-Gubina — APPROVED** [review]

**Review by gtema — APPROVED** [review]

**Pull request merged**